libraries/chain/controller.cpp

```diff
@@ -3484,6 +3464,69 @@ struct controller_impl {
       pending->push();
    }
 
+   void log_applied(controller::block_status s) const {
+      if (replaying) // fork_db_root_block_num not available during replay
+         return;
+      fc::time_point now = fc::time_point::now();
+      // if syncing and not current block, then only report every 1000 blocks
+      if (now - chain_head.timestamp() > fc::minutes(5) && chain_head.block_num() % 1000 != 0)
+         return;
+
+      const auto& br = pending->_block_report;
+      if (s == controller::block_status::incomplete) {
+         const auto& new_b = chain_head.block();
+         ilog("Produced block ${id}... #${n} @ ${t} signed by ${p} "
+              "[trxs: ${count}, lib: ${lib}, confirmed: ${confs}, net: ${net}, cpu: ${cpu}, elapsed: ${et}, time: ${tt}]",
+              ("id", chain_head.id().str().substr(8, 16))("n", new_b->block_num())("p", new_b->producer)("t", new_b->timestamp)
+              ("count", new_b->transactions.size())("lib", fork_db_root_block_num())("confs", new_b->confirmed)
+              ("net", br.total_net_usage)("cpu", br.total_cpu_usage_us)("et", br.total_elapsed_time)
+              ("tt", now - br.start_time));
+
+         if (_update_produced_block_metrics) {
+            produced_block_metrics metrics;
+            metrics.subjective_bill_account_size_total = subjective_bill.get_account_cache_size();
+            metrics.scheduled_trxs_total = db.get_index<generated_transaction_multi_index, by_delay>().size();
+            metrics.trxs_produced_total = new_b->transactions.size();
+            metrics.cpu_usage_us = br.total_cpu_usage_us;
+            metrics.total_elapsed_time_us = br.total_elapsed_time.count();
+            metrics.total_time_us = (now - br.start_time).count();
+            metrics.net_usage_us = br.total_net_usage;
+            metrics.last_irreversible = chain_head.irreversible_blocknum();
+            metrics.head_block_num = chain_head.block_num();
+            _update_produced_block_metrics(metrics);
+         }
+         return;
+      }
+
+      ilog("Received block ${id}... #${n} @ ${t} signed by ${p} " // "Received" instead of "Applied" so it matches existing log output
```

Comment:
It might break tools but more keeping it `Received/Produced` as it is the expected output people expect.

The values of `block_status` is only needed to distinguish between `Received` and `Produced`.